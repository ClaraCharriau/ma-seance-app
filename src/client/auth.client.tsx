import { User } from '../models/User';
import axios from 'axios';

const user_auth_path = 'http://localhost:7878/auth';
const user_verify_path = 'http://localhost:7878/verify';
const user_sign_in_path = 'http://localhost:7878/sign-in';

export const loginUser = async (email: string, password: string): Promise<User> => {
    return await axios
        .post(user_auth_path, {
            email: email,
            password: password
        })
        .then(response => {
            console.log('user-auth-client-response POST user :', response.data);
            return response.data;
        })
        .catch(error => {
            console.error('Error: could not log user : ' + error.message);
            throw error;
        });
};

export const checkAccountExists = async (email: string): Promise<boolean> => {
    return await axios
        .post(user_verify_path, {
            email: email
        })
        .then(function (response) {
            console.log('user-verify-client-response POST verify :', response.data);
            if (response.data && typeof response.data.exists === 'boolean') {
                return response.data.exists;
            }
            throw Error('Could not parse api /verify response. ');
        })
        .catch(function (error: Error) {
            console.error('Error: could not verify if user exists' + error.message);
            throw error;
        });
};

export const signIn = async (pseudo: string, email: string, password: string): Promise<User> => {
    return await axios
        .post(user_sign_in_path, {
            pseudo: pseudo,
            email: email,
            password: password
        })
        .then(function (response) {
            console.log('user-sign-in-client-response POST user :', response.data);
            return response.data;
        })
        .catch(function (error: Error) {
            console.error('Error: could not sign in user' + error.message);
            throw error;
        });
};
