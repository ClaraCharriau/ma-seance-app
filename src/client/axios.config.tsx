import axios, { AxiosResponse } from 'axios';
import { toast } from 'react-toastify';
import config from '../config/config.helper';
import { UserToken } from '../model/UserToken';

/**
 * Set axios global configuration
 */
export const axiosInstance = axios.create({
    baseURL: config.services.prefix,
    headers: {
        'Content-Type': 'application/json'
    }
});

/**
 * Set authentication header if token is found
 */
axiosInstance.interceptors.request.use(config => {
    const maSeanceIdToken = localStorage.getItem('maSeanceId');
    if (maSeanceIdToken) {
        const parsedUsertoken: UserToken = JSON.parse(maSeanceIdToken);
        config.headers.Authorization = `Bearer ${parsedUsertoken.access_token}`;
    }
    return config;
});

/**
 * Set global behavior for responses
 */
axiosInstance.interceptors.response.use(
    response => handleResponse(response),
    error => handleError(error)
);

export const handleError = (error: any) => {
    if (error.response.status === 401 || error.response.status === 403) {
        throw new Response(error.response.data, {
            status: error.response.status,
            statusText: error.response.statusText
        });
    }
    toast.error("Une erreur s'est produite.");
    if (error.response) {
        throw new Response(error.response.data, {
            status: error.response.status,
            statusText: error.response.statusText
        });
    }
};

export const handleResponse = (response: AxiosResponse) => {
    return response.data;
};
