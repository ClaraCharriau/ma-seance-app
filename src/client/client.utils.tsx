import { AxiosResponse } from 'axios';

export const HOST = 'http://localhost:7878';

export const handleError = (error: any) => {
    if (error.response) {
        throw new Response(error.response.data, {
            status: error.response.status,
            statusText: error.response.statusText
        });
    }
    console.error(error.message);
    throw error;
};

export const handleResponse = (response: AxiosResponse) => {
    return response.data;
};

export const handleDeleteResponse = (response: AxiosResponse) => {
    return response;
};
