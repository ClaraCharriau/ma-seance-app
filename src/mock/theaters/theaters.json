[
    {
        "id": "1",
        "name": "C2L Saint-Germain",
        "address": "25-27-29, rue du Vieux-Marche 78100 Saint-Germain-en-Laye",
        "imagePath": "/c2l-saint-germain",
        "bookingPath" : "https://www.ugc.fr"
    },
    {
        "id": "2",
        "name": "C2L Poissy",
        "address": "112 Rue du Général de Gaulle 78300 Poissy",
        "imagePath": "/c2l-poissy",
        "bookingPath" : "https://www.ugc.fr"
    },
    {
        "id": "3",
        "name": "UGC Ciné Cité La Défense",
        "address": "Le Dôme - Centre commercial Westfield - Les 4 Temps 92092 Puteaux",
        "imagePath": "/ugc-defense-puteaux",
        "bookingPath" : "https://www.ugc.fr"
    },
    {
        "id": "4",
        "name": "Les 5 Caumartin",
        "address": "101, rue Saint-Lazare 75009 Paris 9e arrondissement",
        "imagePath": "/5-caumartin-paris",
        "bookingPath" : "https://www.cinqcaumartin.com"
    },
    {
        "id": "5",
        "name": "UGC Ciné Cité Les Halles",
        "address": "7 Place de la Rotonde 75001 Paris",
        "imagePath": "/ugc-les-halles-paris",
        "bookingPath" : "https://www.ugc.fr"
    },
    {
        "id": "6",
        "name": "UGC Normandie",
        "address": "116 bis Av. des Champs-Élysées 75008 Paris",
        "imagePath": "/ugc-normandie-paris",
        "bookingPath" : "https://www.ugc.fr"
    },
    {
        "id": "7",
        "name": "Publicis Cinémas",
        "address": "129, av. des Champs-Elysées 75008 Paris 8e arrondissement",
        "imagePath": "/publicis-paris",
        "bookingPath" : "https://www.publiciscinemas.com/films/"
    },
    {
        "id": "8",
        "name": "Elysées Lincoln",
        "address": "14 Rue Lincoln 75008 Paris",
        "imagePath": "/lincoln-paris",
        "bookingPath" : "https://www.lelincoln.com"
    },
    {
        "id": "9",
        "name": "Les 3 Luxembourg",
        "address": "67 Rue Monsieur le Prince, 75006 Paris",
        "imagePath": "/3-luxembourg-paris",
        "bookingPath" : "https://www.lestroisluxembourg.com"
    },
    {
        "id": "10",
        "name": "La Filmothèque du Quartier Latin",
        "address": "9 Rue Champollion, 75005 Paris",
        "imagePath": "/filmotheque-paris",
        "bookingPath" : "https://www.lafilmotheque.fr"
    },
    {
        "id": "11",
        "name": "Le Champo - Espace Jacques Tati",
        "address": "51, rue des Ecoles 75005 Paris",
        "imagePath": "/champo-paris",
        "bookingPath" : "https://www.cinema-lechampo.com"
    },
    {
        "id": "12",
        "name": "Reflet Médicis",
        "address": "3 Rue Champollion, 75005 Paris",
        "imagePath": "/reflet-medicis-paris",
        "bookingPath" : "https://dulaccinemas.com/cinema/2950/reflet-medicis/seances"
    },
    {
        "id": "13",
        "name": "Écoles Cinéma Club",
        "address": "23, rue des Ecoles 75005 Paris",
        "imagePath": "/ecole-cinema-paris",
        "bookingPath" : "https://pariscinemaclub.com/ecoles-cinema-club/"
    },
    {
        "id": "14",
        "name": "Le Grand Action",
        "address": "5 Rue des Écoles, 75005 Paris",
        "imagePath": "/grand-action-paris",
        "bookingPath" : "https://www.legrandaction.com"
    },
    {
        "id": "15",
        "name": "MK2 Bibliothèque",
        "address": "128-162 avenue de France 75013 Paris",
        "imagePath": "/mk2-bibliotheque-paris",
        "bookingPath" : "https://www.mk2.com/salle/mk2-bibliotheque"
    },
    {
        "id": "16",
        "name": "Pathé Angers",
        "address": "1, avenue des Droits de l'Homme 49000 Angers",
        "imagePath": "/pathe-angers",
        "bookingPath" : "https://www.pathe.fr/cinemas/cinema-pathe-angers"
    },
    {
        "id": "17",
        "name": "Les 400 Coups",
        "address": "12, rue jeanne Moreau 49000 Angers",
        "imagePath": "/400-coups-angers",
        "bookingPath" : "https://www.les400coups.org"
    },
    {
        "id": "18",
        "name": "UGC Ciné Cité Strasbourg Etoile",
        "address": "25, avenue du Rhin 67100 Strasbourg",
        "imagePath": "/ugc-etoile-strasbourg",
        "bookingPath" : "https://www.ugc.fr"
    },
    {
        "id": "19",
        "name": "Mégarex",
        "address": "route du Rhin-za du taubenhof 67500 Haguenau",
        "imagePath": "/megarex-haguenau",
        "bookingPath" : "https://www.megarex.fr/pages/a-laffiche-3.html"
    },
    {
        "id": "20",
        "name": "Le Cosmos",
        "address": "3, rue des Francs-Bourgeois 67000 Strasbourg",
        "imagePath": "/cosmos-strasbourg",
        "bookingPath" : "https://cinema-cosmos.eu"
    },
    {
        "id": "21",
        "name": "Star Saint-Exupery",
        "address": "18, rue du 22-Novembre 67000 Strasbourg",
        "imagePath": "/star-st-ex-strasbourg",
        "bookingPath" : "https://www.cinema-star.com/"
    },
    {
        "id": "22",
        "name": "Star",
        "address": "27, rue du Jeu des enfants 67000 Strasbourg",
        "imagePath": "/star-strasbourg",
        "bookingPath" : "https://www.cinema-star.com/"
    },
    {
        "id": "23",
        "name": "Cinéma Vox",
        "address": "17, rue des Francs-Bourgeois 67000 Strasbourg",
        "imagePath": "/vox-strasbourg",
        "bookingPath" : "https://www.cine-vox.com"
    }
]
