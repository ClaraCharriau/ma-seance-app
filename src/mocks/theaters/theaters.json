[
  {
    "id": 1,
    "name": "C2L Saint-Germain",
    "address": "25-27-29, rue du Vieux-Marche 78100 Saint-Germain-en-Laye",
    "imgPath": "/c2l-saint-germain"
  },
  {
    "id": 2,
    "name": "C2L Poissy",
    "address": "112 Rue du Général de Gaulle 78300 Poissy",
    "imgPath": "/c2l-poissy"
  },
  {
    "id": 3,
    "name": "UGC Ciné Cité La Défense",
    "address": "Le Dôme - Centre commercial Westfield - Les 4 Temps 92092 Puteaux",
    "imgPath": "/ugc-defense-puteaux"
  },
  {
    "id": 4,
    "name": "Les 5 Caumartin",
    "address": "101, rue Saint-Lazare 75009 Paris 9e arrondissement",
    "imgPath": "/5-caumartin-paris"
  },
  {
    "id": 5,
    "name": "UGC Ciné Cité Les Halles",
    "address": "7 Place de la Rotonde 75001 Paris",
    "imgPath": "/ugc-les-halles-paris"
  },
  {
    "id": 6,
    "name": "UGC Normandie",
    "address": "116 bis Av. des Champs-Élysées 75008 Paris",
    "imgPath": "/ugc-normandie-paris"
  },
  {
    "id": 7,
    "name": "Publicis Cinémas",
    "address": "129, av. des Champs-Elysées 75008 Paris 8e arrondissement",
    "imgPath": "/publicis-paris"
  },
  {
    "id": 8,
    "name": "Elysées Lincoln",
    "address": "14 Rue Lincoln 75008 Paris",
    "imgPath": "/lincoln-paris"
  },
  {
    "id": 9,
    "name": "Les 3 Luxembourg",
    "address": "67 Rue Monsieur le Prince, 75006 Paris",
    "imgPath": "/3-luxembourg-paris"
  },
  {
    "id": 10,
    "name": "La Filmothèque du Quartier Latin",
    "address": "9 Rue Champollion, 75005 Paris",
    "imgPath": "/filmotheque-paris"
  },
  {
    "id": 11,
    "name": "Le Champo - Espace Jacques Tati",
    "address": "51, rue des Ecoles 75005 Paris",
    "imgPath": "/champo-paris"
  },
  {
    "id": 12,
    "name": "Reflet Médicis",
    "address": "3 Rue Champollion, 75005 Paris",
    "imgPath": "/reflet-medicis-paris"
  },
  {
    "id": 13,
    "name": "Écoles Cinéma Club",
    "address": "23, rue des Ecoles 75005 Paris",
    "imgPath": "/ecole-cinema-paris"
  },
  {
    "id": 14,
    "name": "Le Grand Action",
    "address": "5 Rue des Écoles, 75005 Paris",
    "imgPath": "/grand-action-paris"
  },
  {
    "id": 15,
    "name": "MK2 Bibliothèque",
    "address": "128-162 avenue de France 75013 Paris",
    "imgPath": "/mk2-bibliotheque-paris"
  },
  {
    "id": 16,
    "name": "Pathé Angers",
    "address": "1, avenue des Droits de l'Homme 49000 Angers",
    "imgPath": "/pathe-angers"
  },
  {
    "id": 17,
    "name": "Les 400 Coups",
    "address": "12, rue jeanne Moreau 49000 Angers",
    "imgPath": "/400-coups-angers"
  },
  {
    "id": 18,
    "name": "UGC Ciné Cité Strasbourg Etoile",
    "address": "25, avenue du Rhin 67100 Strasbourg",
    "imgPath": "/ugc-etoile-strasbourg"
  },
  {
    "id": 19,
    "name": "Mégarex",
    "address": "route du Rhin-za du taubenhof 67500 Haguenau",
    "imgPath": "/megarex-haguenau"
  },
  {
    "id": 20,
    "name": "Le Cosmos",
    "address": "3, rue des Francs-Bourgeois 67000 Strasbourg",
    "imgPath": "/cosmos-strasbourg"
  },
  {
    "id": 21,
    "name": "Star Saint-Exupery",
    "address": "18, rue du 22-Novembre 67000 Strasbourg",
    "imgPath": "/star-st-ex-strasbourg"
  },
  {
    "id": 22,
    "name": "Star",
    "address": "27, rue du Jeu des enfants 67000 Strasbourg",
    "imgPath": "/star-strasbourg"
  },
  {
    "id": 23,
    "name": "Cinéma Vox",
    "address": "17, rue des Francs-Bourgeois 67000 Strasbourg",
    "imgPath": "/vox-strasbourg"
  }
]
