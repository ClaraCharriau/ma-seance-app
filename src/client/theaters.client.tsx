import axios from 'axios';
import { HOST, handleError, handleResponse } from './client.utils';
import { Theater } from '../models/Theater';

const theaters_path = HOST + '/theaters';

export const getTheaterById = async (theaterId: number): Promise<Theater> => {
    return await axios
        .get(theaters_path + '/' + theaterId)
        .then(response => handleResponse(response))
        .catch(error => handleError(error));
};
