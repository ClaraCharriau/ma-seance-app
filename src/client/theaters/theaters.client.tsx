import axios from 'axios';
import { HOST, handleError, handleResponse } from '../client.utils';
import { Theater } from '../../models/Theater';
import { Movie } from '../../models/Movie';
import { MovieScreenings } from '../../models/MovieScreenings';

const THEATERS_PATH = HOST + '/theaters';
const SCREENINGS_PATH = '/screenings';

export const getTheaterById = async (theaterId: string): Promise<Theater> => {
    return await axios
        .get(THEATERS_PATH + '/' + theaterId)
        .then(response => handleResponse(response))
        .catch(error => handleError(error));
};

export const getTheaterMoviesByTheaterId = async (theaterId: string): Promise<Movie[]> => {
    return await axios
        .get(THEATERS_PATH + '/' + theaterId + '/movies')
        .then(response => handleResponse(response))
        .catch(error => handleError(error));
};

export const getMovieScreeningsByTheaterIdAndDay = async (
    theaterId: string,
    day: string
): Promise<MovieScreenings[]> => {
    const params = {
        day
    };
    return await axios
        .get(THEATERS_PATH + '/' + theaterId + SCREENINGS_PATH, { params })
        .then(response => handleResponse(response))
        .catch(error => handleError(error));
};
