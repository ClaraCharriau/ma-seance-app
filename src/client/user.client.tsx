import { Theater } from '../models/Theater';
import axios from 'axios';
import { handleError, handleResponse } from './client.utils';

const host = 'http://localhost:7878';
const users_path = host + '/users';

export const getUserFavTheaters = async (id: number): Promise<Theater[]> => {
    return await axios
        .get(users_path + '/' + id + '/fav-theaters')
        .then(response => handleResponse(response))
        .catch(error => handleError(error, 'GET /fav-theaters'));
};
