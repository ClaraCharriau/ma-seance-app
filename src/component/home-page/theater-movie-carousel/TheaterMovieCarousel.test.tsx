import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import TheaterMovieCarousel from './TheaterMovieCarousel';
import mockMovies from '../../../mocks/theaters/theater-movies.json';
import { getTheaterMoviesByTheaterId } from '../../../client/theaters/theaters.client';

jest.mock('../../../client/theaters/theaters.client', () => ({
    getTheaterMoviesByTheaterId: jest.fn()
}));
const mockGetTheaterMovies = getTheaterMoviesByTheaterId as jest.MockedFunction<typeof getTheaterMoviesByTheaterId>;
describe('Theater movie carousel component tests', () => {
    test('should render component', () => {
        // Given
        mockGetTheaterMovies.mockResolvedValue(mockMovies);
        // When
        const component = render(
            <BrowserRouter>
                <TheaterMovieCarousel theaterId={'1'} />
            </BrowserRouter>
        );
        // then
        expect(component.baseElement).toMatchSnapshot();
    });

    test('should fail to render component', () => {
        // Given
        mockGetTheaterMovies.mockRejectedValue(new Error());
        // When
        const component = render(
            <BrowserRouter>
                <TheaterMovieCarousel theaterId={'1'} />
            </BrowserRouter>
        );
        // then
        expect(component.baseElement).toMatchSnapshot();
    });
});
