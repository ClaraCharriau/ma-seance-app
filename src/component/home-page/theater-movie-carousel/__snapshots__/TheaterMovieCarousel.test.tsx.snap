// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theater movie carousel component tests should fail to render component 1`] = `
<body>
  <div>
    <div
      class="carousel carousel"
    >
      <div
        class="spinner"
      />
    </div>
  </div>
</body>
`;

exports[`Theater movie carousel component tests should render component 1`] = `
<body>
  <div>
    <div
      class="carousel carousel"
    >
      <div
        class="spinner"
      />
    </div>
  </div>
</body>
`;
